src/events/external_clients.py

```diff
@@ -59,58 +64,92 @@ def __init__(self, charm: "EtcdOperatorCharm"):
         )
 
         self.framework.observe(self.etcd_provides.on.mtls_cert_updated, self._on_mtls_cert_updated)
+        self.framework.observe(
+            self.etcd_provides.on.bulk_resources_requested, self._on_bulk_resources_requested
+        )
         self.framework.observe(
             self.charm.on[EXTERNAL_CLIENTS_RELATION].relation_broken, self._on_relation_broken
         )
 
-    def _on_mtls_cert_updated(self, event: MtlsCertUpdatedEvent[RequirerCommonModel]) -> None:  # noqa: C901
-        """Handle the ca chain updated event."""
-        if not self.charm.state.cluster.model or not self.charm.state.unit_server.model:
-            logger.error("peer data not available")
+    def _on_bulk_resources_requested(  # noqa: C901
+        self, event: BulkResourcesRequestedEvent[RequirerCommonModel]
+    ) -> None:
+        """Handle bulk resources requested event."""
+        prevent_reason = self._exists_preventing_reason()
```

Comment:
```suggestion
        preventing_reason = self._exists_preventing_reason()
```

(also in subsequent usages and definitions)